topdown/crypto.go

```diff
@@ -205,6 +209,55 @@ func builtinCryptoX509ParseRSAPrivateKey(_ BuiltinContext, operands []*ast.Term,
 	return iter(ast.NewTerm(value))
 }
 
+func builtinCryptoParsePrivateKey(_ BuiltinContext, operands []*ast.Term, iter func(*ast.Term) error) error {
+
+	a := operands[0].Value
+	input, err := builtins.StringOperand(a, 1)
+	if err != nil {
+		return err
+	}
+
+	// get the raw private key
+	rawKey, err := getRSAPrivateKeyFromString(string(input))
+	if err != nil {
+		return err
+	}
+	key, err := parsePrivateKey(rawKey)
```

Comment:
question: what's to stop us skipping this step and marshalling any value returned by `getRSAPrivateKeyFromString`?

Reply:
you're right. Marshalling it directly without any casting achieves the exact same thing. I'll remove the function and fix tests.